lib/statsd/instrument/rubocop/metric_value_keyword_argument.rb

```diff
@@ -1,35 +1,35 @@
 # frozen-string-literal: true
 
+require_relative '../rubocop' unless defined?(RuboCop::Cop::StatsD)
+
 module RuboCop
   module Cop
     module StatsD
       # This Rubocop will check for providing the value for a metric using a keyword argument, which is
       # deprecated. Use the following Rubocop invocation to check your project's codebase:
       #
       #    rubocop --require \
-      #      `bundle show statsd-instrument`/lib/statsd/instrument/rubocop/metric_value_keyword_argument.rb \
+      #      `bundle show statsd-instrument`/lib/statsd/instrument/rubocop.rb \
       #      --only StatsD/MetricValueKeywordArgument
       #
       # This cop will not autocorrect offenses. Most of the time, these are easy to fix by providing the
       # value as the second argument, rather than a keyword argument.
       #
       # `StatsD.increment('foo', value: 3)` => `StatsD.increment('foo', 3)`
       class MetricValueKeywordArgument < Cop
-        MSG = 'Do not use the value keyword argument, but use a positional argument'
+        include RuboCop::Cop::StatsD
 
-        STATSD_METRIC_METHODS = %i{increment gauge measure set histogram distribution key_value}
+        MSG = 'Do not use the value keyword argument, but use a positional argument'
 
         def on_send(node)
-          if node.receiver&.type == :const && node.receiver&.const_name == "StatsD"
-            if STATSD_METRIC_METHODS.include?(node.method_name)
-              last_argument = if node.arguments.last&.type == :block_pass
-                node.arguments[node.arguments.length - 2]
-              else
-                node.arguments[node.arguments.length - 1]
-              end
-
-              check_keyword_arguments_for_value_entry(node, last_argument) if last_argument&.type == :hash
+          if metric_method?(node)
+            last_argument = if node.arguments.last&.type == :block_pass
+              node.arguments[node.arguments.length - 2]
+            else
+              node.arguments[node.arguments.length - 1]
             end
+
+            check_keyword_arguments_for_value_entry(node, last_argument) if last_argument&.type == :hash
```

Comment:
Why not use the new `keyword_arguments` you added to the `RuboCop::Cop::StatsD` module.

```suggestion
            keyword_args = keyword_arguments(node)
            check_keyword_arguments_for_value_entry(node, keyword_args) if keyword_args
```

Reply:
Fixed